### PR TITLE
Adding an IAM Amazon-service extension

### PR DESCRIPTION
**Pull request by oscerd**

### CODE REVIEW

File: docs/src/main/asciidoc/amazon-iam.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+It's a major rewrite of the 1.x code base that offers two programming models (Blocking & Async).
+
+include::./status-include.adoc[]
+
+The Quarkus extension supports two programming models:
+
+* Blocking access using URL Connection HTTP client (by default) or the Apache HTTP Client
+* https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/basics-async.html[Asynchronous programming] based on JDK's `CompletableFuture` objects and the Netty HTTP client.
+
+include::{generated-dir}/config/quarkus-amazon-sqs.adoc[opts=optional, leveloffset=+1]
```
Comment (gsmet) [inline docs/src/main/asciidoc/amazon-iam.adoc:25]: This is incorrect.

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted a small issue.

@marcinczeczko could you have a look at this one?

File: docs/src/main/asciidoc/amazon-iam.adoc (hunk 1 above)
Comment (oscerd) [reply docs/src/main/asciidoc/amazon-iam.adoc:25]: Yeah, I'll update.

Comment (oscerd) [reply docs/src/main/asciidoc/amazon-iam.adoc]: It should be ok now.

**Comment by sberyozkin** [comment]
@oscerd is it related to #10663 ? Thanks

**Comment by oscerd** [comment]
@sberyozkin no, it's a different service. 

https://aws.amazon.com/iam/?nc1=h_ls
https://aws.amazon.com/secrets-manager/?nc1=h_ls

**Comment by oscerd** [comment]
Any feedback on this?

**Comment by marcinczeczko** [comment]
From my perspective it looks fine.

**Comment by oscerd** [comment]
I think this should be ok to merge

**Comment by sberyozkin** [comment]
@oscerd Hi, a quick question, how does it sync with the current authenticated Quarkus user ? I.e, can anyone who has authenticated call IAM ? thanks

**Comment by oscerd** [comment]
> @oscerd Hi, a quick question, how does it sync with the current authenticated Quarkus user ? I.e, can anyone who has authenticated call IAM ? thanks

Authenticated to AWS? What do you mean with authenticated Quarkus user? Thanks

**Comment by sberyozkin** [comment]
> Authenticated to AWS? What do you mean with authenticated Quarkus user? Thanks

No, the user who has authenticated to the Quarkus endpoint (via Basic Auth or OIDC) which makes an outbound call to AWS IAM.
For example, what if a user access a Quarkus endpoint after authenticating via AWS SSO (I haven't tried but `quarkus-oidc` would handle it).

**Comment by oscerd** [comment]
You'll always need to use the AWS credentials to interact with IAM service.
The IAM exposed API are for management, so you can create a user or add a
group etc

Il gio 8 ott 2020, 13:02 sberyozkin <notifications@github.com> ha scritto:

**Comment by oscerd** [comment]
So you won't authenticated through it. It's just an exposed API for doing stuff related to the IAM service

**Comment by sberyozkin** [comment]
@oscerd 
> So you won't authenticated through it. It's just an exposed API for doing stuff related to the IAM service
I understand, thanks, the question was, can any authenticated Quarkus user access this IAM service :-) ?

**Comment by oscerd** [comment]
I think so.

**Comment by sberyozkin** [comment]
@oscerd Thanks, yeah, that looks similar to a service account approach with Google Cloud services which Loic is working upon.

That is fine, going forward, if it is possible with IAM, we can link AWS SSO + IAM with `quarkus-oidc` (something which is also proposed for the Google services), so only the users who authenticate into Quarkus via AWS SSO can invoke. Ping us please if it will be of interest

**Review by sberyozkin — APPROVED** [review]
LGTM, but waiting for Guillaume to double check as well

**Comment by sberyozkin** [comment]
@oscerd Please squash the commits

**Comment by oscerd** [comment]
@sberyozkin done

**Comment by sberyozkin** [comment]
@oscerd thanks, waiting for @gsmet to confirm the requested change looks good for him.
In meantime, please also rebase and force push as Stuart has done some fixes to do with timeouts (JDK8 build is failing here)

**Review by gsmet — APPROVED** [review]

**Pull request merged**